pages/accountLists/[accountListId]/reports/financialAccounts/[[...financialAccount]].page.test.tsx

```diff
@@ -0,0 +1,160 @@
+import { FinancialAccountQuery } from 'src/components/Reports/FinancialAccountsReport/Context/FinancialAccount.generated';
+import { FinancialAccountsQuery } from 'src/components/Reports/FinancialAccountsReport/FinancialAccounts/FinancialAccounts.generated';
+import { FinancialAccountsMock } from 'src/components/Reports/FinancialAccountsReport/FinancialAccounts/FinancialAccountsMocks';
+import i18n from 'src/lib/i18n';
+import theme from 'src/theme';
+import FinancialAccountsPage from './[[...financialAccount]].page';
+
+const accountListId = 'account-list-1';
+const financialAccountId = 'financialAccountId';
+const defaultRouter = {
+  query: { accountListId },
+  isReady: true,
+};
+
+const Components = ({ router = defaultRouter }: { router?: object }) => (
+  <I18nextProvider i18n={i18n}>
+    <LocalizationProvider dateAdapter={AdapterLuxon}>
+      <SnackbarProvider>
+        <ThemeProvider theme={theme}>
+          <TestRouter router={router}>
+            <GqlMockedProvider<{
+              FinancialAccountSummary: FinancialAccountSummaryQuery;
+              FinancialAccount: FinancialAccountQuery;
+              FinancialAccounts: FinancialAccountsQuery;
+            }>
+              mocks={{
+                FinancialAccountSummary: defaultFinancialAccountSummary,
+                FinancialAccount: defaultFinancialAccount,
+                ...FinancialAccountsMock,
+              }}
+            >
+              <FinancialAccountsPage />
+            </GqlMockedProvider>
+          </TestRouter>
+        </ThemeProvider>
+      </SnackbarProvider>
+    </LocalizationProvider>
+  </I18nextProvider>
+);
+
+describe('Financial Accounts Page', () => {
+  it('should show initial financial accounts page', async () => {
+    const { findByText } = render(<Components />);
+
+    expect(await findByText('Responsibility Centers')).toBeInTheDocument();
+
+    expect(await findByText('Test Account')).toBeInTheDocument();
+  });
+
+  it('should show the summary page for a financial account', async () => {
+    const { findByText, findByRole, getByText, queryByText } = render(
+      <Components
+        router={{
+          query: { accountListId, financialAccount: [financialAccountId] },
+          isReady: true,
+        }}
+      />,
+    );
+
+    expect(await findByText('Account 1')).toBeInTheDocument();
+
+    expect(queryByText('Responsibility Centers')).not.toBeInTheDocument();
+
+    expect(
+      await findByRole('heading', { name: 'Category' }),
+    ).toBeInTheDocument();
+
+    expect(getByText('Opening Balance')).toBeInTheDocument();
+  });
+
+  it('should show the transactions page for a financial account', async () => {
+    const { findByText, findByRole, getByText, queryByText, queryByRole } =
+      render(
+        <Components
+          router={{
+            query: {
+              accountListId,
+              financialAccount: [financialAccountId, 'entries'],
+            },
+            isReady: true,
+          }}
```

Comment:
This router is used a few times. You could consider extracting it into an `entriesRouter`.

```ts
const entriesRouter = {
  ...defaultRouter,
  query: {
    ...defaultRouter.query,
    financialAccount: [financialAccountId, 'entries'],
  },
};
```

Reply:
Good catch